injectproxy/routes.go

```diff
@@ -206,7 +222,12 @@ func NewRoutes(upstream *url.URL, label string, opts ...Option) (*routes, error)
 
 func (r *routes) enforceLabel(h http.HandlerFunc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		lvalue := req.FormValue(r.label)
+		lvalue, err := r.getLabelValue(req)
+		if err != nil {
+			prometheusAPIError(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
 		if lvalue == "" {
```

Comment:
I guess we can remove this test since `getLabelValue()` always returns a non-empty string (it should probably be highlighted in the comments and/or function name).

Reply:
I think it can still return an empty string when both the `r.labelValue` and form value are empty. But we handle the empty case there which would remove the need to check for empty at the call site. Wdyt?

Reply:
you're correct, I misread the code :)
what you suggested sounds good to me